string command
int64 pattern_no
---
bool success
